pkg/store/bucket.go

```diff
@@ -1472,10 +1472,15 @@ func (b *bucketBlock) readChunkRange(ctx context.Context, seq int, off, length i
 	}
 	defer runutil.CloseWithLogOnErr(b.logger, r, "readChunkRange close range reader")
 
-	if _, err = io.Copy(buf, r); err != nil {
+	copied, err := io.Copy(buf, r)
+	if err != nil {
 		b.chunkPool.Put(c)
 		return nil, errors.Wrap(err, "read range")
 	}
+	if copied != len {
+		b.chunkPool.Put(c)
+		return nil, errors.Errorf("Read less than specified wanted=%v got=%v", len, copied)
```

Comment:
We should begin it with a lowercase letter. Also, I think this improves the formatting:

```suggestion
		return nil, errors.Errorf("read %v bytes but got %v", len, copied)
```